src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,45 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
-    public void getStatistic(String fromFileName, String toFileName) {
+    private static final String COMMA_SEPARATOR = ",";
+    private static final String BUY_KEYWORD = "buy";
 
+    public void getStatistic(String fromFileName, String toFileName) {
+        StringBuilder stringBuilder = new StringBuilder();
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            String str = bufferedReader.readLine();
+            while (str != null) {
+                stringBuilder.append(str).append(COMMA_SEPARATOR);
+                str = bufferedReader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read the file: ", e);
+        }
+        String[] stringsFromFile = stringBuilder.toString().split(COMMA_SEPARATOR);
+        int supplyAllSum = 0;
+        int buyAllSum = 0;
+        for (int i = 0; i < stringsFromFile.length - 1; i += 2) {
+            if (stringsFromFile[i].equals(BUY_KEYWORD)) {
+                buyAllSum += Integer.parseInt(stringsFromFile[i + 1]);
+            } else {
+                supplyAllSum += Integer.parseInt(stringsFromFile[i + 1]);
+            }
+        }
```

Comment:
So, as I see, first you are reading all lines and saving them to `StringBuilder` (lines 14 - 23), then you are processing them again and calculating the `supply` and `buy` sum. Could we make it easier?

Can we calculate the sum on the lines 14-23, without saving to StringBuilder?